### PR TITLE
fix(trading): fills fees maker discounts

### LINKED ISSUE #5401: Fees and discounts for makers are not calculating correctly
## The Chore

We seem to be inferring the maker / taker role based on the `aggressor` and whether the connected party is the buyer or seller. 

Then if we are taker things are fine but if we're maker we're flipping the sign on the maker fee to show that you are being paid.  In isolation thats OK... but then the total fee ends up being infra + liq - maker which is wrong. 

We also have a problem that we cant tell from the trades API if a market is in auction, when the fees should change.  We've asked Core for an enhancement for that ... but there's a short term workaround below.

These issues impact both the total fee / discount in the table AND the values in the tooltip.

image

### Fees

**In Continuous trading**

if Taker:
makerFee = makerFee
infraFee = infraFee
liqFee = liqFee
Total fee = makerFee + infraFee + liqFee

If Maker:
makerFee = - makerFee
infraFee = 0
liqFee = 0
Total fee = - makerFee

**If in Auction (which we can detect by checking both buyer AND seller makerFee = 0)**

makerFee = 0
infraFee = infraFee / 2
liqFee = liqFee / 2
Total fee = (infraFee + liqFee)/2

### Discounts (these apply to both volume and referal discounts)

**In Continuous trading**

if Taker:
makerFeeDiscount = makerFeeDiscount
infraFeeDiscount = infraFeeDiscount
liqFeeDiscount = liqFeeDiscount
Total Discount= makerFeeDiscount + infraFeeDiscount + liqFeeDiscount

If Maker:
makerFeeDiscount = makerFeeDiscount
infraFeeDiscount = 0
liqFeeDiscount = 0
Total discount = as above

**If in Auction (which we can detect by checking both buyer AND seller makerFee = 0)**

makerFeeDiscount = 0
infraFeeDiscount = infraFeeDiscount / 2
liqFeeDiscount = liqFeeDiscount / 2
Total fee Discount = (infraFeeDiscount + liqFeeDiscount)/2

## Tasks

- [ ]
- [ ]

### PR DESCRIPTION
**Pull request by MadalinaRaicu**
# Related issues 🔗

Closes #5401 

# Description ℹ️

- [x] Refactor fills table 
- [x] Override fills discounts if the aggressor is the maker

# Demo 📺

Screenshot 2023-11-30 at 18 09 02

Screenshot 2023-11-30 at 18 09 08

# Technical 👨‍🔧

Details of technical implementation that reviewers may need to be aware of, if applicable.

### CODE REVIEW

File: libs/fills/src/lib/fills.utils.ts
Comment (mattrussell36) [inline libs/fills/src/lib/fills.utils.ts:1]: Can we just call this fills-utils.ts or just utils.ts to keep convention with source files

File: libs/fills/src/lib/fills.utils.ts [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+  let {
+    makerFeeVolumeDiscount,
+    makerFeeReferralDiscount,
+    infrastructureFeeReferralDiscount,
+    infrastructureFeeVolumeDiscount,
+    liquidityFeeReferralDiscount,
+    liquidityFeeVolumeDiscount,
+  } = fees;
+
+  if (isMarketActive) {
```
Comment (mattrussell36) [inline libs/fills/src/lib/fills.utils.ts:87]: Mentioned in the other PR but we should make sure this is covered in the unit tests for this function

**Review by mattrussell36 — APPROVED** [review]

File: libs/fills/src/lib/fills.utils.ts
Comment (MadalinaRaicu) [reply libs/fills/src/lib/fills.utils.ts:1]: Updated

**Comment by dalebennett1992** [comment]
Checked the fees for maker and taker, all looks to be ok. Ben checked suspended yesterday. have checked values with discount, and continuous trading fees

**Review by dalebennett1992 — APPROVED** [review]

**Pull request merged**